tests/unit/Expressions/Literals/IntegerTest.php

```diff
@@ -0,0 +1,94 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of php-cypher-dsl.
+ *
+ * Copyright (C) Wikibase Solutions
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace WikibaseSolutions\CypherDSL\Tests\Unit\Expressions\Literals;
+
+use PHPUnit\Framework\TestCase;
+use TypeError;
+use WikibaseSolutions\CypherDSL\Expressions\Literals\Integer;
+use WikibaseSolutions\CypherDSL\Types\PropertyTypes\IntegerType;
+
+/**
+ * @covers \WikibaseSolutions\CypherDSL\Expressions\Literals\Integer
+ */
+final class IntegerTest extends TestCase
+{
+    public function testZero(): void
+    {
+        $decimal = new Integer(0);
+
+        $this->assertSame("0", $decimal->toQuery());
+        $this->assertEquals('0', $decimal->getValue());
+    }
+
+    public function testInstanceOfIntegerType(): void
+    {
+        $this->assertInstanceOf(IntegerType::class, new Integer(0));
+    }
+
+    public function testThrowsTypeErrorOnInvalidType(): void
+    {
+        $this->expectException(TypeError::class);
+
+        new Integer([]);
+    }
+
+    public function testThrowsTypeErrorOnInvalidString(): void
+    {
+        $this->expectException(TypeError::class);
+
+        new Integer('not an integer');
+    }
+
+    /**
+     * @dataProvider provideToQueryData
+     *
+     * @param $number
+     */
+    public function testToQuery($number, string $expected): void
+    {
+        $decimal = new Integer($number);
+
+        $this->assertSame($expected, $decimal->toQuery());
+        $this->assertEquals($expected, $decimal->getValue());
+    }
+
+    /**
+     * @dataProvider provideInvalidInputData
+     */
+    public function testInvalidInput($input): void
+    {
+        $this->expectException(TypeError::class);
+        new Integer($input);
+    }
+
+    public function provideToQueryData(): array
+    {
+        return [
+            [1, "1"],
+            [2, "2"],
+            ["2147483649", "2147483649"],
+            ["9223372036854775816", "9223372036854775816"],
+            [-12, "-12"],
+            [69, "69"],
+            ["292922929312312831203129382304823043284234729847294324724982749274294729427429471230918457", "292922929312312831203129382304823043284234729847294324724982749274294729427429471230918457"],
```

Comment:
Neo4j/cypher does not handle numbers larger than `9223372036854775816` iirc. Do you want to test these?

Reply:
This is supported by the DSL, so I think it is fine to include a test for it. In the future, we could prohibit users from using numbers that large, but since the 5.0.0 release is feature-frozen, it will need to be moved to 6.0.0.